### PR TITLE
VMware: add new module : vmware_host_lockdown

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                elif not host.config.adminDisabled:
+                    if desired_state == 'present':
+                        host.EnterLockdownMode()
+                        results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                    else:
+                        results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
+                        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
+                changed = True
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py]: @Akasurde 
```
if host.config.adminDisabled:
    if desired_state == 'absent':
        host.ExitLockdownMode()
        results['host_lockdown_state'][host.name]['previous_state'] = 'present'
        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
    else:
        results['host_lockdown_state'][host.name]['previous_state'] = 'present'
        results['host_lockdown_state'][host.name]['current_state'] = 'present'
elif not host.config.adminDisabled:
    if desired_state == 'present':
        host.EnterLockdownMode()
        results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
        results['host_lockdown_state'][host.name]['current_state'] = 'present'
    else:
        results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
changed = True
```
should be:
```
changed = False
if host.config.adminDisabled:
    results['host_lockdown_state'][host.name]['previous_state'] = 'present'
    if desired_state == 'absent':
        host.ExitLockdownMode()
        changed = True
        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
    else:
        results['host_lockdown_state'][host.name]['current_state'] = 'present'
else:
    results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
    if desired_state == 'present':
        host.EnterLockdownMode()
        changed = True
        results['host_lockdown_state'][host.name]['current_state'] = 'present'
    else:
        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
```

What do you think ?

**Review by tchernomax — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py]: OK.

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py]: Done.

File: lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                    else:
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                elif not host.config.adminDisabled:
+                    results['host_lockdown_state'][host.name]['previous_state'] = 'absent'
+                    if desired_state == 'present':
+                        host.EnterLockdownMode()
+                        results['host_lockdown_state'][host.name]['current_state'] = 'present'
+                    else:
+                        results['host_lockdown_state'][host.name]['current_state'] = 'absent'
+                changed = True
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_lockdown.py]: You did half of what I suggested :)
`changed = False` should be at the top, and `changed = True` should follow host.ExitLockdownMode and ost.EnterLockdownMode.

**Review by tchernomax — CHANGES_REQUESTED** [review]
ping @Akasurde

**Review by tchernomax — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@tchernomax Thanks for review.

**Pull request merged**